ecl/hql/hqlgram.y

```diff
@@ -10560,6 +10566,46 @@ optSortList
     | ',' sortList
     ;
 
+optTraceFlags
+    : traceFlags
+    |                   {   $$.setNullExpr(); }
+    ;
+
+traceFlags
+    : ',' traceFlag     {   $$.setExpr($2.getExpr()); }
+    | traceFlags ',' traceFlag
+                        {   $$.setExpr(createComma($1.getExpr(), $3.getExpr())); }
+    ;
+
+traceFlag
+    : KEEP expression   {
+                            parser->normalizeExpression($2, type_int, false);
+                            $$.setExpr(createExprAttribute(keepAtom, $2.getExpr()));
+                        }
+    | SKIP expression   {
+                            parser->normalizeExpression($2, type_int, false);
+                            $$.setExpr(createExprAttribute(skipAtom, $2.getExpr()));
+                        }
+    | SAMPLE expression   {
+                            parser->normalizeExpression($2, type_int, false);
+                            $$.setExpr(createExprAttribute(sampleAtom, $2.getExpr()));
+                        }
+    | NAMED '(' constExpression ')'
+                        {
+                            parser->normalizeStoredNameExpression($3);
+                            $$.setExpr(createExprAttribute(namedAtom, $3.getExpr()));
+                            $$.setPosition($1);
+                        }
+    | expression
+                        {
+                            //MORE:SORTLIST  Allow a sortlist to be expanded!
+                            parser->normalizeExpression($1);
```

Comment:
check boolean?

    parser->normalizeExpression($1, type_boolean, false);

I don't think there is any need for the convertAll call.